codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -282,6 +285,16 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                 return;
             }
 
+            boolean isInformational = false;
+            if ((!connection.isServer() &&
+                    (isInformational = HttpStatusClass.valueOf(headers.status()) == INFORMATIONAL)
+                    && stream.isInformationalHeadersReceived()) ||
```

Comment:
_Practical_! Bah! Such high expectations! :smile: I'll say in _practice_ they're pretty unheard of. But as far as semi-practical use-cases:

1. Multiple 1xx status codes. You con conceive of another 1xx status code you may want to mix with 100... say [WebDav's 102 Processing](https://tools.ietf.org/html/rfc2518#section-10.1). You send 100 to tell the client you have resources available, send 102 to say you have received the request and are working on the response, and then 200 when you done.

2. Sending 102 multiple times looks useful. It's supposed to tell the client not to time out yet. If it took a _long_ time to process you could understand sending that multiple times. Granted, in that case you may want to rearchitect. But it would be practical...

3. I had thought _someone_ would be using 100 Continue multiple times as a sort of checkpointing system. It turns out [103 Checkpoint](https://web.archive.org/web/20151013212135/http://code.google.com/p/gears/wiki/ResumableHttpRequestsProposal) is a quasi-thing. [There](http://advanced-servlets.yahoogroups.narkive.com/9TbXg1mU/sending-multiple-100-continue-responses-before-a-200-ok) [is](https://stackoverflow.com/questions/22818059/several-100-continue-received-from-the-server) [evidence](https://github.com/php-mod/curl/issues/19) that _somebody_ is sending 100 multiple times, but none of them explain why. And it definitely seems rare.